core/base/src/TSystem.cxx

```diff
@@ -3374,7 +3377,13 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
          if (!keep) k->SetBit(kMustCleanup);
          fCompiled->Add(k);
 
-         gInterpreter->GetSharedLibDeps(library);
+         if (!linkDepLibraries) {
+            // if linkDepLibraries the library might not know its dependencies
+            // and "just load the library" might fail.
+            std::string depLibraries = gInterpreter->GetSharedLibDeps(library);
+            for (auto lib : ROOT::Split(depLibraries, " "))
```

Comment:
We seem to have that functionality in `ForeachSharedLibDep`. Can you use it?

Reply:
Ouch, thanks for pointing that out (this hid an actual problem).